packages/core/src/domain/context/customerDataTracker.ts

```diff
@@ -0,0 +1,93 @@
+import { ONE_KIBI_BYTE, computeBytesCount } from '../../tools/utils/byteUtils'
+import { throttle } from '../../tools/utils/functionUtils'
+import type { Context } from '../../tools/serialisation/context'
+import { jsonStringify } from '../../tools/serialisation/jsonStringify'
+import { display } from '../../tools/display'
+import { CustomerDataType } from './contextConstants'
+
+// RUM and logs batch bytes limit is 16KB
+// ensure that we leave room for other event attributes and maintain a decent amount of event per batch
+// (3KB (customer data) + 1KB (other attributes)) * 4 (events per batch) = 16KB
+export const CUSTOMER_DATA_BYTES_LIMIT = 3 * ONE_KIBI_BYTE
+
+// We observed that the compression ratio is around 8 in general, but we also want to keep a margin
+// because some data might not be compressed (ex: last view update on page exit). We chose 16KiB
+// because it is also the limit of the 'batchBytesCount' that we use for RUM and Logs data, but this
+// is a bit arbitrary.
+export const CUSTOMER_COMPRESSED_DATA_BYTES_LIMIT = 16 * ONE_KIBI_BYTE
+
+export const BYTES_COMPUTATION_THROTTLING_DELAY = 200
+
+export type CustomerDataTracker = ReturnType<typeof createCustomerDataTracker>
+export const enum CustomerDataCompressionStatus {
+  Unknown,
+  Enabled,
+  Disabled,
+}
+
+export function createCustomerDataTracker(
+  type: CustomerDataType,
+  compressionStatus = CustomerDataCompressionStatus.Disabled
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏FMU it is the current behaviour but would not it worth having a single count for all customer data rather than one by type?
for example, currently customer data could be FF 2k + User 2k + global 2k and no warn would be generated, right?